web-ide/proxy/static/css/webide.main.css

```diff
@@ -40,24 +40,38 @@ div.window-appicon {
 .part.titlebar .menubar div.menubar-menu-button[aria-label="File"] .action-item .menubar-menu-items-holder .actions-container[aria-label="File"] li.action-item:nth-child(8) { /*empty*/
   display: none !important;
 }  
+/**************************************************************************************************/
+
+
+/*It is not possible to disable all references to terminal, debugger, and other functionality we wish to hide.
+ For one, vscode uses a the same (or identical) high level div element for different contextual menus. This means we can't hide
+ via nth-child because the same html has different menu content based on context. Instead we remove as many references as we can
+ and hide the front end functionality.
+*/
 
-/*remove from global actions menu*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(1), /*command pallete*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(2), /*empty*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(4), /*Extensions*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(5), /*Keyboard shortcuts*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(6), /*empty*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(7), /*User Snippets*/
-.monaco-menu-container .monaco-action-bar ul.actions-container li.action-item:nth-child(8) { /*empty*/
+.global-activity { /*hide the entire global actions menu, this is mostly unwanted functionality*/
   display: none !important;
 }
 
+/*hide the commands widget, this is useful for file navigation but entering '>' allows you to execute any command*/
+.monaco-quick-open-widget {
```

Comment:
I'd rather leave this present and accept that it's a possibility to sneak stuff though. The file navigation is just too useful. Many of the commands (like converting case etc) are so standard dev features that without the quick open widget things aren't much fun.

Reply:
👍